drivers/auxdisplay/auxdisplay_serlcd.c

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright (c) 2023 Jan Henke <Jan.Henke@taujhe.de>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sparkfun_serlcd
+
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/auxdisplay.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(auxdisplay_serlcd, CONFIG_AUXDISPLAY_LOG_LEVEL);
+
+/*
+ * | in ASCII, used to begin a display command
+ */
+#define SERLCD_BEGIN_COMMAND 0x7C
+
+/*
+ * special command for the underlying display controller
+ */
+#define SERLCD_BEGIN_SPECIAL_COMMAND 0xFE
+
+union auxdisplay_display_state {
+	uint8_t byte;
+	struct {
+		uint8_t padding: 5;
+		bool power: 1;
+		bool cursor: 1;
+		bool blinking: 1;
+	} __packed;
```

Comment:
Something like
```suggestion
	struct __packed {
#ifdef CONFIG_LITTLE_ENDIAN
		uint8_t padding: 5;
		bool power: 1;
		bool cursor: 1;
		bool blinking: 1;
#else
		bool blinking: 1;
		bool cursor: 1;
		bool power: 1;
		uint8_t padding: 5;
#endif
	};
```

Reply:
Applied.

As a side note: This is the first time for me that *bit ordering* is referred to as endianess, which commonly refers to *byte ordering*.